src/Client/Request.php

```diff
@@ -71,8 +72,8 @@ public function xmlContent()
      */
     public function arguments(): array
     {
-        $xml = SoapXml::fromString($this->xmlContent());
-        $arguments = Arr::first(XMLSerializer::domNodeToArray($xml->getBody()));
+        $doc = Document::fromXmlString($this->xmlContent());
+        $arguments = Arr::first(XMLSerializer::domNodeToArray($doc->locate(new SoapBodyLocator())));
```

Comment:
Maybe worth the try, veewee/xml - which is inluded now, also provides a xml to array command:
https://github.com/veewee/xml/blob/master/docs/encoding.md#xml_decode

Reply:
Absolutly right...yeah included it, but it seems i missed some parts. Yeah maybe i can get rid finally of my custom xml helpers ^^

Reply:
maybe something like this?

```php
use VeeWee\Xml\Dom\Document;
use function VeeWee\Xml\Dom\Configurator\loader;
use function VeeWee\Xml\Dom\Configurator\pretty_print;
use function VeeWee\Xml\Dom\Loader\xml_string_loader;
use function VeeWee\Xml\Dom\Mapper\xml_string;
use function VeeWee\Xml\Encoding\xml_decode;

$doc = Document::configure(
    pretty_print(),
    loader(xml_string_loader($this->xmlContent()))
);

$firstNode = $doc->locate(new SoapBodyLocator())?->firstElementChild;
if (!$firstNode) {
    return [];
}

return xml_decode(xml_string()($firstNode));
```

Running namespace optimizations doesnt feel quite good here, since it is CPU intensive.

Reply:
The "problem" was that i need the namespaces removed. So that i get clean arguments. My approach delivered:
``
array:2 [
  "ns1:prename" => "Corona"
  "ns1:lastname" => "Pandemic"
]
``

too much operations ^^.....trying your new code 👍

Reply:
Tried your code: 
````
VeeWee\Xml\Encoding\Exception\EncodingException : XML issues detected: 
[ERROR] : Namespace prefix SOAP-ENV on node is not defined
 (201) on line 1,15
[ERROR] : Namespace prefix SOAP-ENV on prename is not defined
 (201) on line 2,20
[ERROR] : Namespace prefix SOAP-ENV on lastname is not defined
 (201) on line 3,21
 /home/dredd/PhpstormProjects/codedredd/laravel-soap/vendor/veewee/xml/src/Xml/Encoding/Exception/EncodingException.php:32
 /home/dredd/PhpstormProjects/codedredd/laravel-soap/vendor/veewee/xml/src/Xml/Encoding/Internal/wrap_exception.php:24
 /home/dredd/PhpstormProjects/codedredd/laravel-soap/vendor/veewee/xml/src/Xml/Encoding/xml_decode.php:27
 /home/dredd/PhpstormProjects/codedredd/laravel-soap/src/Client/Request.php:71
```` 

this namespaces ^^

Reply:
Yeah they are declared above..
For cases where you don't care about namespaces and prefixes,
It might make sense to create a RemoveNamespaces visitor for the traverser that can be used whilst decoding.

That functionality is not available ATM though

Reply:
Well....thats what i thought ^^....but i hoped i have maybe overseen a configurator fitting my case. Thinking about it.
Anyways thanks for the input & review! ❤️  Never thought i get your attention 😄

Reply:
No problem! I'm keeping an eye on some SOAP repo's. Let me know if there is anything you would want my help with.

About above ... couldn't resist...

https://github.com/veewee/xml/releases/new?tag=v1.3.0

```php
use VeeWee\Xml\Dom\Document;
use VeeWee\Xml\Dom\Traverser\Visitor\RemoveNamespaces;
use function VeeWee\Xml\Dom\Configurator\traverse;
use function VeeWee\Xml\Encoding\element_decode;

$doc = Document::fromXmlString(<<<EOXML
<?xml version="1.0"?>

<soap:Envelope
xmlns:soap="http://www.w3.org/2003/05/soap-envelope/"
soap:encodingStyle="http://www.w3.org/2003/05/soap-encoding">

<soap:Body xmlns:m="http://www.example.org/stock">
  <m:GetStockPrice>
    <m:StockName>IBM</m:StockName>
  </m:GetStockPrice>
</soap:Body>

</soap:Envelope>
EOXML
);

$method = $doc->xpath()->querySingle('//soap:Body')->firstElementChild;

var_dump(element_decode($method, traverse(new RemoveNamespaces())));
```

Output
```
array(1) {
  'GetStockPrice' =>
  array(1) {
    'StockName' =>
    string(3) "IBM"
  }
}
```

Reply:
You are my man 💯 
But i get that error now:
````
VeeWee\Xml\Exception\RuntimeException : Failed querying XPath query: //soap:Body
[ERROR] : Undefined namespace prefix
 (1219) on line 0,0
 /home/dredd/PhpstormProjects/codedredd/laravel-soap/vendor/veewee/xml/src/Xml/Exception/RuntimeException.php:33
 /home/dredd/PhpstormProjects/codedredd/laravel-soap/vendor/veewee/xml/src/Xml/ErrorHandling/disallow_issues.php:35
 /home/dredd/PhpstormProjects/codedredd/laravel-soap/vendor/azjezz/psl/src/Psl/Result/Failure.php:85
 /home/dredd/PhpstormProjects/codedredd/laravel-soap/vendor/veewee/xml/src/Xml/ErrorHandling/disallow_issues.php:36
 /home/dredd/PhpstormProjects/codedredd/laravel-soap/vendor/veewee/xml/src/Xml/Dom/Xpath/Locator/query_single.php:32
 /home/dredd/PhpstormProjects/codedredd/laravel-soap/vendor/veewee/xml/src/Xml/Dom/Xpath.php:60
 /home/dredd/PhpstormProjects/codedredd/laravel-soap/vendor/veewee/xml/src/Xml/Dom/Xpath.php:79
 /home/dredd/PhpstormProjects/codedredd/laravel-soap/src/Client/Request.php:61

````

But it works now that way:
````php
$doc = Document::fromXmlString($this->xmlContent());
$method = $doc->locate(new SoapBodyLocator())?->firstElementChild;
$arguments = element_decode($method, traverse(new RemoveNamespaces()));
````

Reply:
Jep, that is the way.
The soap prefix can be different based on the XML. Hence the soap body locator uses the soap preset for xpath.